pytest-embedded-idf/pytest_embedded_idf/idf_unity_tester.py

```diff
@@ -0,0 +1,452 @@
+import logging
+import time
+from functools import wraps
+from threading import Semaphore, Thread
+from typing import List, Union
+
+from pexpect.exceptions import TIMEOUT
+from pytest_embedded import Dut, unity, utils
+from pytest_embedded_idf.dut import UnittestMenuCase
+
+DEFAULT_RETRY_COUNT = 3
+DEFAULT_TIMEOUT = 90
+
+
+def timeit(func):
+    @wraps(func)
+    def timeit_wrapper(*args, **kwargs):
+        start_time = time.perf_counter()
+        result = func(*args, **kwargs)
+        end_time = time.perf_counter()
+        total_time = end_time - start_time
+        logging.info(f'Testcase "{args[1].name}" took {total_time:.5f} seconds')
```

Comment:
logging doesn't help. should raise a timeout and terminate running the current test case.

Reply:
I agree with Raising timeout. My plan was to use this line later to track test execution time in the junit report.

Reply:
this timeout would also help kill those hanging processes. that's why raise immediately is better, IMHO